lib/puppet/resource/type_collection.rb

```diff
@@ -49,11 +49,6 @@ def inspect
     }.inspect
   end
 
-  def <<(thing)
```

Comment:
Will put it pack and mark the entire TypeCollection as being @api private in 5.0.0. Most of the methods in that class should never ever be called by anyone outside of the puppet code base. For this particular method we can add deprecations but not for the others unless we start looking at where calls are coming from.... (and we won't). Eventually I want to see TypeCollection gone from the compiler all together.